consensus/src/dag/tests/order_rule_tests.rs

```diff
@@ -187,3 +187,83 @@ proptest! {
         }
     }
 }
+
+#[test]
+fn test_order_rule_basic() {
+    let dag = vec![
+        vec![Some(vec![]), Some(vec![]), Some(vec![]), Some(vec![])],
+        vec![
+            Some(vec![false, true, true, true]),
+            Some(vec![true, true, true, false]),
+            Some(vec![false, true, true, true]),
+            None,
+        ],
+        vec![
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+        ],
+        vec![
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            Some(vec![true, false, true, true]),
+            None,
+        ],
+        vec![
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            None,
+        ],
+        vec![
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            Some(vec![true, true, true, false]),
+            None,
+        ],
+    ];
+    let (_, validator_verifier) = random_validator_verifier(4, None, false);
+    let validators = validator_verifier.get_ordered_account_addresses();
+    let author_indexes = validator_verifier.address_to_validator_index().clone();
+    let nodes = generate_dag_nodes(&dag, &validators);
+    let epoch_state = Arc::new(EpochState {
+        epoch: 1,
+        verifier: validator_verifier,
+    });
+    let mut dag = Dag::new(epoch_state.clone(), Arc::new(MockStorage::new()));
+    for round_nodes in &nodes {
+        for node in round_nodes.iter().flatten() {
+            dag.add_node(node.clone()).unwrap();
+        }
+    }
+    let display = |node: &NodeMetadata| (node.round(), *author_indexes.get(node.author()).unwrap());
+    let dag = Arc::new(RwLock::new(dag.clone()));
+    let (mut order_rule, mut receiver) = create_order_rule(epoch_state.clone(), dag);
+    for node in nodes.iter().flatten().flatten() {
+        order_rule.process_new_node(node);
+    }
+    let expected_order = vec![
+        // anchor (1, 0) has 1 votes, anchor (3, 1) has 2 votes and a path to (1, 0)
+        vec![(1, 0)],
+        // anchor (2, 1) has 3 votes
+        vec![(1, 2), (1, 1), (2, 1)],
+        // anchor (3, 1) has 2 votes
+        vec![(1, 3), (2, 2), (2, 0), (3, 1)],
+        // anchor (4, 2) has 3 votes
+        vec![(3, 3), (3, 2), (3, 0), (4, 2)],
+        // anchor (5, 2) has 3 votes
+        vec![(4, 1), (4, 0), (5, 2)],
```

Comment:
same round authors is reversed. its deterministic but not intuitive. can we add a comment somewhere in order_rule that this is the expected order.

Reply:
yeah it's because we reverse the order